fuel-vm/src/interpreter/executors/main.rs

```diff
@@ -106,6 +223,16 @@ where
         let root = contract.root();
         let storage_root = Contract::initial_state_root(storage_slots.iter());
         let id = contract.id(salt, &root, &storage_root);
+        let mut cumulative_predicate_gas: Word = 0;
+
+        // Calculate cumulative predicate gas for inputs
+        for input in create.inputs() {
```

Comment:
Why duplicate this calculation & logic here when it's already known from the `Checked<Create>` `metadata().gas_used_by_predicates`?

Reply:
Because we don't have a `Checked` transaction during execution=) We can try to store it inside the `Interpreter` but I think it is simpler to calculate it than propagate